### ISSUE TITLE
Would love to see some updates on this.

### ISSUE THREAD

**Issue by Iomega0318** [SimonOmega/UOPatchServer#1]
This looks very promising and would be great for all free shards alike. I myself have been collecting patches for ages from ~4.0.10-7.0.24 up until the cut off the patch server anyways lol.. I do have them available online as well and would love to set up a patch server for all to use, however at a rate of 1kbs it would take years to transfer the data :)

**Comment by SimonOmega** [comment]
I have had a lot go on in my life the past year.  If you see it as valuable I will try to finish it and make a full fledged server.

**Comment by Nylar1971** [comment]
Well, I would appreciate it tons :) which means there are probably a few
silent people out there that would like to see it as well. I encourage you
to finish it and I bet the response is more than you might expect :)

Thanks for your interest!

Doug

**Comment by Iomega0318** [comment]
I agree, and yes that would be so amazing if you could work on it! I would be willing to either host the patch files on my server as well, or pay for some hosting online. Either way I am sure a lot more of the community than just us want to see this :)

and yes, thank you for coming back!

David

**Comment by SimonOmega** [comment]
Started looking at some things.  If I can sort out the client version data being sent, I can pragmatically figure out the proper patch list for varying versions.  That way I am not dumping everything in the archive to the client. This will take some time and testing.

I only have 7.0.10 to 7.0.23.3 that i can test.  I lost the rest of my patches, so I can check around the web for some.  Looky here: https://code.google.com/p/download-client/downloads/list

@Iomega0318 Maybe we can setup a google code repo to hold the patches.

I am still trying to test the padding in the PatchDataRequest segment clients send.  See if it actually is continuing a patch from a stopped point.

Good news is the way it is right now, I already have it threading so multiple clients can connect and it can handout patches.  I forgot I had it that far.  It shouldn't clash when multiple clients request the same file.

More to come later....

**Comment by SimonOmega** [comment]
I am working on it slowly but surely.

**Comment by Iomega0318** [comment]
By all means take your time, I will give it some tests when I get everything set back up.. Work has me pretty busy at the moment as well as life, grr! Until then and sorry for getting back to you so late I have patches going all the way back to 4.0.10a-7.0.24.2 with an old AoS installer.

You can find the installer here:
<a href="http://www.uofreeshards.net/attachments/ultima-online-aos-4-0-9-0b-exe.450/">Ultima-Online-AoS-4.0.9.0b.exe</a>

And the patches here:
<a href="http://www.uofreeshards.net/attachments/aos2d_win32_4-0-10a-aos2d_win32_4-0-11f-tar-gz.713/">aos2d_win32_4.0.10a-aos2d_win32_4.0.11f.tar.gz</a>
<a href="http://www.uofreeshards.net/attachments/aos2d_win32_5-0-0a-aos2d_win32_5-0-65-tar-gz.714/">aos2d_win32_5.0.0a-aos2d_win32_5.0.65.tar.gz</a>
<a href="http://www.uofreeshards.net/attachments/aos2d_win32_6-0-0-0-uosa_win32_6-0-14-4-tar-gz.715/">aos2d_win32_6.0.0.0-uosa_win32_6.0.14.4.tar.gz</a>
<a href="http://www.uofreeshards.net/attachments/uosa_win32_7-0-0-0-uosa_win32_7-0-24-2-tar-gz.716/">uosa_win32_7-0-0-0-uosa_win32_7-0-24-2.tar.gz</a>

Don't remember if you need to register or not still haven't found time to set all the permissions, if you do let me know since it's set for admin approval..

**Comment by Iomega0318** [comment]
Other than that it sounds like your doing great so far! I can't wait to play with it some more!!

**Comment by Nylar1971** [comment]
Hell yeah buddy, very nice!

**Comment by Nylar1971** [comment]
Heh looks like I am a member of the site, just can't get to the files yet.
Ah well, It looks like things are really moving along, congrats o your
great strides so far!

**Comment by Iomega0318** [comment]
Hehe I try lol, I approved you by the way. It's set up for admin approval since I still haven't finished with the permissions, just been procrastinating lol.. I had some even older patch files on my old server but the motherboard went bad, and unfortunately it used SCSI drives.. So other than buying a new server I haven't found a way to get the files off the drives..

**Comment by SimonOmega** [comment]
I have been busy lately.  Class and work.  I have a plan for the patch list. It will be the biggest issue/change.  The others will be small.  Bare with me, I am still on it.  Installing Fedora 21 Workstation, and Windows just updated and killed my mouse.  To bed for tonight.

**Comment by SimonOmega** [comment]
Rewrote a few things tonight.

**Comment by SimonOmega** [comment]
No Commit to GitHub yet but I updated #4 I am generating patch list data from the server now.  I have to change the communication code in the send patch data section and the server will intelligently send only the updates the client needs. 

PS...  I tried implementing notifications.  It caused the client to flip out.  So i will investigate that later.

----

CLIENT >> 127.0.0.1 sent Client Hello/Request (b'\x00\x00\x00\x15')

Server Sending: Protocol, Command, and UseSelf
CLIENT >> 127.0.0.1 sent Version Information Win32_UOSA 34

PATCH LIST BABY

SERVER << sent 127.0.0.1 PatchListData entry {'patname': b'uosa_win32_7-0-11-1.pat', 'patnamelen': b'\x00\x00\x00\x17', 'rtpnamelen': b'\x00\x00\x00\x17', 'patlen': b'\x00\x00\x00\r', 'rtpname': b'uosa_win32_7-0-11-1.rtp', 'rtplen': b'\x03s\xa4\x8e'}

SERVER << sent 127.0.0.1 PatchListData entry {'patname': b'uosa_win32_7-0-11-2.pat', 'patnamelen': b'\x00\x00\x00\x17', 'rtpnamelen': b'\x00\x00\x00\x17', 'patlen': b'\x00\x00\x00\r', 'rtpname': b'uosa_win32_7-0-11-2.rtp', 'rtplen': b'\x03\xb76b'}

SERVER << sent 127.0.0.1 PatchListData entry {'patname': b'uosa_win32_7-0-11-3.pat', 'patnamelen': b'\x00\x00\x00\x17', 'rtpnamelen': b'\x00\x00\x00\x17', 'patlen': b'\x00\x00\x00\r', 'rtpname': b'uosa_win32_7-0-11-3.rtp', 'rtplen': b'\x07\xdc\xc7\x1a'}

SERVER << sent 127.0.0.1 PatchListData entry {'patname': b'uosa_win32_7-0-11-4.pat', 'patnamelen': b'\x00\x00\x00\x17', 'rtpnamelen': b'\x00\x00\x00\x17', 'patlen': b'\x00\x00\x00\r', 'rtpname': b'uosa_win32_7-0-11-4.rtp', 'rtplen': b'\x03tnC'}

CLIENT >> 127.0.0.1 sent a Request for File uosa_win32_7-0-11-1.pat ({'padding': 0, 'working_filename': 'uosa_win32_7-0-11-1.pat', 'namelen': 23})

**Comment by SimonOmega** [comment]
Changes to Socket and Threads.  A little more clean.  I think the threads I am using are IO threads and not necessarily Multiprocessing threads.   Just following the recommendations of the PyDocs.  It will be pretty soon.

**Comment by SimonOmega** [comment]
Code just uploaded should be a functional beta.

**Comment by SimonOmega** [comment]
@Iomega0318 Thank you for the patches ^_^ ...  Just I can't access them... o_o  would you like me to help you setup a github,googlecode, dropbox, or such to host them?

@Nylar1971 Thank you as well.  Would you like to Beta test this code for me?  Just because it works for me and my patches does not mean it will work on everyone's system.  ?_?

**Comment by Iomega0318** [comment]
@SimonOmega That is awesome news! I will play around with it when I get some free time this week. Also let me know if you register on my forum and I'll approve you, I really need to finish the permissions on that forum LOL.. As far as setting up a host site goes I would be more than happy to help, Google Code I looked into and it will no longer work the way we would want it to, github or dropbox might be a much better option.
